### PR TITLE
batches: use user name/e-mail with unauthored batch specs

### PR DESCRIPTION
**Pull request by adeola-ak**
fixes the E2E test that failed after merge in #43828 and needed to be reverted

Implementing the same fallback behaviour for changeset specs on SSBC that we have on the client side where we try to get the author name and e-mail for commits from the environment if they're not in the batch spec by pulling that from the user's account details and created a sub-package author to do so

## Test plan
Tested created a bc in various cases with missing information

### CODE REVIEW

File: enterprise/dev/ci/integration/executors/tester/main.go [hunk 1]
```diff
@@ -98,8 +98,8 @@ func main() {
 				Title:             "Hello World",
 				Body:              "My first batch change!",
 				CommitMessage:     "Append Hello World to all README.md files",
-				CommitAuthorName:  "Sourcegraph",
-				CommitAuthorEmail: "batch-changes@sourcegraph.com",
```
Comment (courier-new) [inline enterprise/dev/ci/integration/executors/tester/main.go:102]: Out of curiosity, why is this name/email not the same as the one [further down in the same test](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/enterprise/dev/ci/integration/executors/tester/main.go?L211-214)?

**Review by courier-new — COMMENTED** [review]
Looks good to me, but could we trigger the executor e2e tests for this PR to double-check before we merge?

**Comment by courier-new** [comment]
From your branch locally you can run `sg ci build main-dry-run`. You should see the full buildkite suite with executors e2e kick off from that. 🙂

**Comment by adeola-ak** [comment]
nice, thanks! looks like its [good](https://buildkite.com/sourcegraph/sourcegraph/builds/193703) beside security scans

**Review by courier-new — APPROVED** [review]
Awesome thanks! I'll approve to unblock, but I also think there's no rush on this if you want to wait until after 4.4 cut tomorrow.

**Comment by adeola-ak** [comment]
thank uuu! yes thats what i was thinking, ill just wait

**Pull request merged**